src/app/store/reducers/VehicleTestResultModel.reducers.ts

```diff
@@ -1,20 +1,61 @@
-import { initialVehicleTestResultModelState } from '../../store/state/VehicleTestResultModel.state';
-import { EVehicleTestResultModelActions, VehicleTestResultModelActions } from '../actions/VehicleTestResultModel.actions';
+import { initialVehicleTestResultModelState } from '@app/store/state/VehicleTestResultModel.state';
+import {
+  EVehicleTestResultModelActions, SetSelectedTestResultModelSuccess,
+  VehicleTestResultModelActions
+} from '../actions/VehicleTestResultModel.actions';
 
-export function VehicleTestResultModelReducers(state = initialVehicleTestResultModelState, action: VehicleTestResultModelActions) {
+export function VehicleTestResultModelReducers(
+  state = initialVehicleTestResultModelState,
+  action: VehicleTestResultModelActions
+) {
   switch (action.type) {
     case EVehicleTestResultModelActions.GetVehicleTestResultModelSuccess: {
       return {
         ...state,
-        selectedTestResultModel: action.payload,
-        error: null // clear error message
+        vehicleTestResultModel: action.payload,
+        error: null
       };
     }
 
     case EVehicleTestResultModelActions.GetVehicleTestResultModelFailure: {
       return {
         ...state,
-        error: action.payload  // capture error message
+        error: action.payload
+      };
+    }
+
+    case EVehicleTestResultModelActions.SetCurrentState: {
+      return {
+        ...state,
+        editState: action.editState
+      };
+    }
+
+    case EVehicleTestResultModelActions.UpdateTestResultSuccess: {
+      const updatedVehicleTestResults = state.vehicleTestResultModel.map((testResult) => {
+        if (
```

Comment:
this can be moved into a state selector I believe,
I am not convinced storing logic in a reducer in the appropriate way.
@alleypa @m-ujaffer what's your view on that please?
I would tend to try to have composition as much possible with reducers simply rendering data, any logic should happen in selectors that we could compose to reshape the state the way we want it This will extract out as much logic as possible which could be handled in some utils functions which will be way easier to test than a possible bloated reducer.

Reply:
Yes I agree @JonLazariniDD. Reducers are not meant to be complicated. I think an essential selector could be missing or the need extra functions out